src/algebra/ordered_ring.lean

```diff
@@ -327,16 +371,11 @@ lt_of_not_ge (λ ha, absurd h (mul_nonpos_of_nonneg_of_nonpos ha hb).not_lt)
 lemma neg_of_mul_pos_right (h : 0 < a * b) (ha : a ≤ 0) : b < 0 :=
 lt_of_not_ge (λ hb, absurd h (mul_nonpos_of_nonpos_of_nonneg ha hb).not_lt)
 
-@[priority 100] -- see Note [lower instance priority]
-instance linear_ordered_semiring.to_nontrivial {α : Type*} [linear_ordered_semiring α] :
-  nontrivial α :=
-{ exists_pair_ne := ⟨0, 1, ne_of_lt zero_lt_one⟩ }
-
 /- TODO This theorem ought to be written in the context of `nontrivial` linearly ordered (additive)
 commutative monoids rather than linearly ordered rings; however, the former concept does not
 currently exist in mathlib. -/
 @[priority 100] -- see Note [lower instance priority]
-instance linear_ordered_semiring.to_no_top_order {α : Type*} [linear_ordered_semiring α] :
+instance linear_ordered_semiring.to_no_top_order {α : Type*} [nontrivial α] [linear_ordered_semiring α] :
```

Comment:
The nontrivial assumption is redundant here.

Reply:
```suggestion
instance linear_ordered_semiring.to_no_top_order {α : Type*} [linear_ordered_semiring α] :
```